plugin/path_rotate_creds.go

```diff
@@ -0,0 +1,108 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const (
+	rotateRolePath = "rotate-role/"
+)
+
+func (b *backend) pathRotateCredentials() *framework.Path {
+	return &framework.Path{
+		Pattern: rotateRolePath + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
+			"name": {
+				Type:        framework.TypeString,
+				Description: "Name of the static role",
+			},
+		},
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.CreateOperation: &framework.PathOperation{
+				Callback:                    b.pathRotateCredentialsUpdate,
+				ForwardPerformanceStandby:   true,
+				ForwardPerformanceSecondary: true,
+			},
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback:                    b.pathRotateCredentialsUpdate,
+				ForwardPerformanceStandby:   true,
+				ForwardPerformanceSecondary: true,
+			},
+		},
+
+		HelpSynopsis:    pathRotateCredentialsUpdateHelpSyn,
+		HelpDescription: pathRotateCredentialsUpdateHelpDesc,
+	}
+}
+
+func (b *backend) pathRotateCredentialsUpdate(ctx context.Context, req *logical.Request, fieldData *framework.FieldData) (*logical.Response, error) {
+	cred := make(map[string]interface{})
+
+	config, err := readConfig(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if config == nil {
+		return nil, errors.New("the config is currently unset")
+	}
+
+	roleName := fieldData.Get("name").(string)
+
+	b.credLock.Lock()
+	defer b.credLock.Unlock()
+
+	role, err := b.readRole(ctx, req.Storage, roleName)
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return nil, nil
```

Comment:
Should this case (trying to rotate a non-existent role) instead be returning an error?

Reply:
Indeed its should, good catch!